streaming/src/main/scala/io/gearpump/streaming/task/TaskActor.scala

```diff
@@ -242,8 +237,31 @@ class TaskActor(
       receiveMessage(inputMessage, sender)
     case upstream@ UpstreamMinClock(upstreamClock) =>
       this.upstreamMinClock = upstreamClock
-      val latestMinClock = minClock
-      val update = UpdateClock(taskId, latestMinClock)
+      val minClocks = subscriptions.map { case (id, sub) => sub -> sub.minClock }.toMap
```

Comment:
This change is over-complicated considering what it want to do.

It use **4** rounds! (map, foldLeft, Math.min, foreach)

Really necessary? Please try to use one round to fix this.